server/src/main/java/org/apache/lucene/queries/BlendedTermQuery.java

```diff
@@ -133,11 +134,15 @@ protected void blend(final TermStates[] contexts, int maxDoc, IndexReader reader
                 // we need to find out the minimum sumTTF to adjust the statistics
                 // otherwise the statistics don't match
                 minSumTTF = Math.min(minSumTTF, reader.getSumTotalTermFreq(terms[i].field()));
+                minDocCount = Math.min(minDocCount, reader.getDocCount(terms[i].field()));
             }
         }
         if (maxDoc > minSumTTF) {
             maxDoc = (int) minSumTTF;
         }
```

Comment:
This is probably redundant now.

`minSumTTF` is guaranteed to be `<= minDocCount`.

Reply:
I changed the logic anyway, so we're not modifying `maxDoc` twice anymore. We can leave this check in place.